### PR TITLE
Advanced Filtering

### PR DESCRIPTION
**Pull request by nicholasmueller**
Issue/Task Number: [715 ](https://github.com/omisego/ewallet/issues/715)

# Overview

depends on https://github.com/omisego/ewallet/pull/1066

Advanced filtering as per this design series -> https://gallery.io/projects/MCHbtQVoQ2HCZfyN0ITInFtb/files/MCHbtQVoQ2HCZeFc8VHDAU1d

Also some filters as per the design are not possible so weren't included

# Changes

- configurable filters in one place (`FilterMap.js`) so you can add, remove, tweak in the future
- new react-select dependency for multi select dropdown

# Usage

Requires a lot of testing, each filter on each page as well as the combination of filters.
Navigate pages, add filters, remove filters, remove breadcrumbs, try different use cases.

how to use on other pages -> add <AdvancedFilter> wherever you want the filter crumbs to show and it will pass a `{ matchAll, matchAny }` object to the `onFilter` prop when filters are defined. You can then inject that to the fetcher query on that page (using local state in this case).

### CODE REVIEW

**Comment by mederic-p** [comment]
Overall:
    - Create Date -> Creation date? Also I’d change `Start` `End` to `From` To`
    - I’d prefer to see the filters as a popup that shows under the `Filter` button when you click on it instead of a plain page but need to talk with the designer first
    - The `Yes` `No` choices could be replaced with a tick box (to discuss with designer)
    - Would be nice to have an autocompleted list showing when we start typing on some fields (wallet/user/…)
    - Can we somehow “store” the filter infos in the url (so we can share a link with all filtering info)
	

Transaction filters: 
    - transfer to / specify target don’t list all possible accounts (missing user accounts)
Transaction Request filters: 
    - Received -> should be Receive
    - Confirmation Type -> Should be Requires confirmation
    - Specify Requested Wallet -> Specify Request Wallet
    - Can we use the same dropdown picker as in transactions (the one with more details about accounts)
    - Would be nice to be able to filter by token and amount also
Transaction Consumption filters: 
    - Same as Transaction Request
    - Consumed Date -> Consumption Date

**Comment by jarindr** [comment]
i'm ok on the implementation wise, for the design improvement can deal with it later I think.

File: apps/frontend/assets/src/adminPanelApp/globalStyle.css [hunk 1]
```diff
@@ -37,7 +37,7 @@ body {
   top: 0;
   right: 0;
   bottom: 0;
-  overflow: hidden;
+  overflow-y: scroll;
```
Comment (jarindr) [inline apps/frontend/assets/src/adminPanelApp/globalStyle.css]: the body overflow should always be hidden, the overflow should be handle on modal side.

**Comment by jarindr** [comment]
Update:
- the filter of transfer from/to when typing doesn't change the dropdown selection items.
- Specify Target also doesn't work.

**Comment by nicholasmueller** [comment]
@mederic-p can you please give it another look? i made all the selects searchable and removed the multi-selects (annoying bug in the dependency I am using where you can't dynamically update the dropdown list)

File: apps/frontend/assets/src/omg-uikit/timepicker/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import DateTime from 'react-datetime'
+
+import { Input } from '..'
```
Comment (jarindr) [inline apps/frontend/assets/src/omg-uikit/timepicker/index.js]: no `..` please 😆

File: apps/frontend/assets/src/omg-uikit/datepicker/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import DateTime from 'react-datetime'
+
+import { Input } from '..'
```
Comment (jarindr) [inline apps/frontend/assets/src/omg-uikit/datepicker/index.js]: here too

**Pull request merged**